common/src/commonMain/kotlin/com/xacalet/minesweeper/common/ui/MainScreen.kt

```diff
@@ -0,0 +1,146 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import com.xacalet.minesweeper.common.data.GameRepository
+import com.xacalet.minesweeper.common.model.GameData
+import com.xacalet.minesweeper.common.model.GameState
+import com.xacalet.minesweeper.common.ui.component.BevelType
+import com.xacalet.minesweeper.common.ui.component.ClassicButton
+import com.xacalet.minesweeper.common.ui.component.NumericDisplay
+import com.xacalet.minesweeper.common.ui.component.bevel
+import com.xacalet.minesweeper.common.ui.component.cells.Cell
+import com.xacalet.minesweeper.common.ui.resources.PainterResources
+
+@Composable
+expect fun MainScreen()
+
+@Composable
+internal fun GameBoard(
+    gameData: GameData,
+    gameRepository: GameRepository,
+    onLongClickCell: () -> Unit = {}
+) {
+    Box(Modifier.wrapContentSize()) {
+        Column(
+            modifier = Modifier
+                .width(IntrinsicSize.Max)
+                .background(Color.LightGray)
+                .bevel()
+                .padding(8.dp)
+        ) {
+            val isAnyCellPressed = remember { mutableStateOf(false) }
+            ControlPanel(
+                gameData = gameData,
+                clickRestartButton = gameRepository::resetGame,
+                isPressed = isAnyCellPressed
+            )
+            Spacer(Modifier.size(8.dp))
+            CellGrid(
+                gameData = gameData,
+                clickCell = { x, y -> gameRepository.onCellClick(x, y) },
+                longClickCell = { x, y ->
+                    if (gameRepository.onCellLongClick(x, y)) {
+                        onLongClickCell()
+                    }
+                },
+                cellPressed = { isPressed -> isAnyCellPressed.value = isPressed }
+            )
+        }
+    }
+}
+
+@Composable
+internal fun GameStateImage(isPressed: Boolean, gameState: GameState) {
+    val painter = if (isPressed) {
+        PainterResources.iconSmileySurprised()
+    } else {
+        when (gameState) {
+            GameState.Won -> PainterResources.iconSmileySunglasses()
+            GameState.Lost -> PainterResources.iconSmileyDizzy()
+            else -> PainterResources.iconSmileySmiling()
+        }
+    }
+
+    Image(
+        modifier = Modifier.padding(6.dp),
+        painter = painter,
+        contentDescription = null
+    )
+}
+
+@Composable
+internal fun ControlPanel(
+    gameData: GameData,
+    clickRestartButton: () -> Unit,
+    isPressed: State<Boolean>
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .bevel(type = BevelType.Lowered)
+            .padding(6.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        val timer = gameData.elapsedSeconds.collectAsState(0)
+        NumericDisplay(value = timer.value)
```

Comment:
As I commented in review for `NumericDisplay.kt` file, it'd be better to remove the specific size from the display segment component to make it reusable, and instead, specify the height we expect for a specific use through the modifier parameter. Well, here's where we should specify the height:
```suggestion
        NumericDisplay(
            modifier = Modifier.height(42.dp),
            value = timer.value
        )
```